src/main/java/org/sonatype/goodies/packageurl/PackageUrlBuilder.java

```diff
@@ -105,9 +108,11 @@ public PackageUrlBuilder qualifiers(final String qualifiers) {
   }
 
   public PackageUrlBuilder qualifier(final String key, final String value) {
-    checkNotNull(key);
-    checkNotNull(value);
-    getQualifiers().put(key, value);
+    requireNonNull(key);
+    requireNonNull(value);
+    if (!value.isEmpty()) {
```

Comment:
The spec actually does say: 

>  value cannot be an empty string: a key=value pair with an empty value is the same as no key/value at all for this key

so IMHO, this is correct.

However, as you know, my overall goal is to do the "as-is" mode for the builder and parser.
So, what I decided is that in "as-is" mode, we will have `...&repository=&qualifier=py2` and in regular mode we will drop repository to have `...&qualifier=py2`.

Reply:
@bradcupit the spec disallows qualifier with blank value as @andrewstein points out above.  In parsing I think this has to silently drop, unsure about the builder api here, it may be better to throw IAE instead with a reasonable message?

I'm on the fence about if it should just silently ignore here with direct api usage.